Source/cryptalgo/SecureSocketPort.h

```diff
@@ -72,19 +72,26 @@ namespace Crypto {
             };
 
         public:
+            // Type of underlying socket
+            enum class sockettype_t : uint8_t {
+                   CLIENT_SOCKET
+                ,  SERVER_SOCKET
+            };
+
             Handler(Handler&&) = delete;
             Handler(const Handler&) = delete;
             Handler& operator=(const Handler&) = delete;
 
             template <typename... Args>
-            Handler(SecureSocketPort& parent, Args&&... args)
+            Handler(SecureSocketPort& parent, sockettype_t type, Args&&... args)
                 : Core::SocketPort(args...)
                 , _parent(parent)
                 , _context(nullptr)
                 , _ssl(nullptr)
                 , _callback(nullptr)
-                , _handShaking(CONNECTING) {
-            }
+                , _handShaking(type == sockettype_t::CLIENT_SOCKET ? CONNECTING : ACCEPTING)
+                , _type{type}
```

Comment:
As the sockettype_t type -> _type is not used anwhere, just drop it, it just confuses..